modules/internal/ChapelRange.chpl

```diff
@@ -2350,38 +2454,54 @@ proc _cast(type t, r: range(?)) where isRangeType(t) {
     }
   }
 
-  private proc chpl__rangeAlignType(type idxType) type {
-    if isIntegralType(idxType) {
-      return idxType;
-    } else if isEnumType(idxType) {
-      // TODO: ultimately, this should be an int of the same width as the enum
-      return int;
-    } else {
-      chpl__rangeIdxTypeError(idxType);
-    }
+  // TODO: What would it take to make this private?  Currently DefaultRectangular
+  // relies on it, but could perhaps refer to range.intIdxType instead?
+  pragma "no doc"
+  proc chpl__idxTypeToIntIdxType(type idxType) type {
+    if isEnumType(idxType) then return int; else return idxType;
   }
 
-  private proc chpl__defaultLowBound(type idxType) {
-    if isIntegralType(idxType) {
-      return 1:idxType;
-    } else if isEnumType(idxType) {
-      // TODO: This is only correct for default-numbered enums; need
-      // to generalize
-      return 2:idxType;
-    } else {
-      chpl__rangeTypeError(idxType);
-    }
+  inline proc chpl__intToIdx(type idxType, i: integral, j ...) {
+    const first = chpl__intToIdx(idxType, i);
+    const rest = chpl__intToIdx(idxType, (...j));
+    return (first, (...rest));
   }
 
-  private proc chpl__defaultHighBound(type idxType) {
-    if isIntegralType(idxType) {
-      return 0:idxType;
-    } else if isEnumType(idxType) {
-      // TODO: This is only correct for default-numbered enums; need
-      // to generalize
-      return 1:idxType;
-    } else {
-      chpl__rangeTypeError(idxType);
-    }
+  inline proc chpl__intToIdx(type idxType, i: integral, j: integral) {
```

Comment:
We briefly discussed this offline, but for the record: I think it could help make this change less verbose if we had a method variant of this function such that we did not need to pass 'idxType' every time.

Reply:
I've made this conversion for ranges.  I'm also going to do it for DefaultRectangular and move the multidimensional versions of chpl__intToIdx() over to that module since they're only used there.  Intriguingly, I had to (or arguably, chose to) use different names for the method name than the standalone function because otherwise the calls to the routine in the initializers tried to call the method version which isn't legal.  I think this requires either using a different name or using a fully-qualified call which seems like a mouthful for this context and this (internal) module.  Let me know if you disagree with any of this or have different ideas.